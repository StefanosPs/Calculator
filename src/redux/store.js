import {createStore, applyMiddleware } from 'redux';


import rootReducer from './root-reducer';

// import logger from 'redux-logger';
// const middlewares = [logger];

// const store = createStore(rootReducer, applyMiddleware(...middlewares));

const store = createStore(rootReducer);

export default store;