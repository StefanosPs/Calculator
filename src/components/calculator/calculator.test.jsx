import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import store from  '../../redux/store';
import Calculator from './calculator';
 

it('renders  Calculator', () => {
    const numberKeyFull = renderer.create(<Provider store={store}><Calculator /></Provider>).toJSON();
    expect(numberKeyFull).toMatchSnapshot();
});