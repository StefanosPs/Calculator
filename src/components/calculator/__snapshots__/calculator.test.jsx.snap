// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders  Calculator 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="mt-auto pt-3 pb-3 bg-dark text-white rounded  shadow-sm container"
    style={
      Object {
        "maxWidth": "720px",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="float-right mt-auto col-12"
      >
        <div
          className="switch float-right"
        >
          <h6>
            Change Theme
          </h6>
          <label
            className="switch-label"
          >
            <input
              onClick={[Function]}
              type="checkbox"
            />
            <span
              className="slider round"
            />
          </label>
        </div>
        <textarea
          className="mt-2 text-white bg-transparent border-0 shadow-sm mb-1 form-control"
          id="history"
          readOnly={true}
        />
        <input
          className="mb-2  text-white bg-transparent border-0 shadow-sm mb-1 form-control"
          defaultValue=""
          id="mathfn"
          readOnly={true}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <div
          className="container"
        >
          <div
            className="justify-content-center row"
          >
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                7
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                8
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                9
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                4
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                5
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                6
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                1
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                2
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                3
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                0
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                .
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="col-6"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Divide"
                type="button"
              >
                /
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Îœultiplication"
                type="button"
              >
                X
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Minus,"
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Add"
                type="button"
              >
                +
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Open Parenthesis"
                type="button"
              >
                (
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Close Parenthesis"
                type="button"
              >
                )
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Squared"
                type="button"
              >
                xÂ²
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Square Root"
                type="button"
              >
                âˆšx
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Undo"
                type="button"
              >
                â†¶
              </button>
            </div>
            <div
              className="col-4"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Clear Display"
                type="button"
              >
                C
              </button>
            </div>
            <div
              className="col-8"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <button
                className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-primary btn-block"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                text="Calculate"
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
